### ISSUE TITLE
Bug when comparing two empty collections.

### ISSUE THREAD

**Issue by dmytro-gokun** [dennisdoomen/FluentAssertions#382]
This test

```
        [Test]
        public void EmptyCollectionsBug()
        {
            var actual = new List<string>();
            var expected = new List<string>();
            actual.Should().Equal(expected);
        }
```

Fails with the following message:
```
Expected collection to be equal to {empty}, but found empty collection.
```

I'm using the latest version (4.3.0)

**Comment by dennisdoomen** [comment]
Hmm, you might have found an edge case (although I agree it's not that weird)

**Comment by dmytro-gokun** [comment]
The thing is that it worked just fine with the previous version.

**Comment by dennisdoomen** [comment]
Yes, managed to reproduce it. Apparently there was no unit test for this scenario and something was refactored internally. Will fix it ASAP.

**Comment by dennisdoomen** [comment]
Fixed in 4.3.1
